File: execute/_50A_py.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class _50A_py {
    public static void main(String[] args) {
        String currentFileName = new Exception().getStackTrace()[0].getClassName(), dirPath = "../input/" + currentFileName;
        File dir = new File(dirPath);

        System.out.println("/******************************************  "  + currentFileName + "  ******************************************/");

        if (dir.exists() && dir.isDirectory()) {
            if (dir.listFiles(File::isFile) != null) {
                for (int i = 0; i < dir.listFiles(File::isFile).length; i++) {
                    String inputFilePath = dirPath + "/input" + currentFileName + "_" + i + ".txt";
                    String outputFilePath = "../output/" + currentFileName + "/output" + currentFileName + "_" + i + ".txt";

                    try {
                        List<String> inputLines = Files.readAllLines(Paths.get(inputFilePath));
                        if (!inputLines.isEmpty()) {

  /******************************************** Answer code ************************************************/
                            // Split the first line by spaces and parse the numbers
                            String[] numbers = inputLines.get(0).trim().split("\\s+");
                            if (numbers.length >= 2) {
                                int num1 = Integer.parseInt(numbers[0]);
                                int num2 = Integer.parseInt(numbers[1]);
                                int result = (num1 * num2) / 2;

 /******************************************** End of Answer code ********************************************/
                                Utils.commonFunction(outputFilePath , result , i);
                            } else {
                                System.err.println("Not enough numbers in input file: " + inputFilePath);
                                System.exit(1);
                            }
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    } catch (NumberFormatException e) {
                        System.err.println("Invalid number format in file: " + inputFilePath);
                        System.exit(1);
                    }
                }
            }
        }
    }
}

File: execute/_50A_py.kt
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths

object _50A_py {
    @JvmStatic
    fun main(args: Array<String>) {
        val currentFileName = Exception().stackTrace[0].className
        val dirPath = "../input/$currentFileName"
        val dir = File(dirPath)

        println("/******************************************  $currentFileName  ******************************************/")

        if (dir.exists() && dir.isDirectory) {
            val files = dir.listFiles { file -> file.isFile }
            files?.forEachIndexed { index, _ ->
                val inputFilePath = "$dirPath/input$currentFileName_$index.txt"
                val outputFilePath = "../output/$currentFileName/output$currentFileName_$index.txt"

                try {
                    val inputLines = Files.readAllLines(Paths.get(inputFilePath))
                    if (inputLines.isNotEmpty()) {
                        // Split the first line by spaces and parse the numbers
                        val numbers = inputLines[0].trim().split("\\s+".toRegex())
                        if (numbers.size >= 2) {
                            val num1 = numbers[0].toInt()
                            val num2 = numbers[1].toInt()
                            val result = (num1 * num2) / 2

                            // Replace Utils.commonFunction with your output handling logic
                            Utils.commonFunction(outputFilePath, result, index)
                        } else {
                            println("Not enough numbers in input file: $inputFilePath")
                            System.exit(1)
                        }
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: NumberFormatException) {
                    println("Invalid number format in file: $inputFilePath")
                    System.exit(1)
                }
            }
        }
    }
}
